pages/myposts.vue

```diff
@@ -412,78 +193,76 @@ onMounted(() => {
   })
 })
 
-const busyWanteds = ref(true)
-const showOldWanteds = ref(false)
-
-const wanteds = computed(() => {
-  return (
-    messages
-      ?.filter(
-        (message) =>
-          message.type === 'Wanted' &&
-          (showOldOffers.value || !message.hasoutcome)
-      )
-      .sort(sortPosts) ?? []
-  )
-})
+/// //////////////////////////////////////////////
+// `posts` holds both OFFERs and WANTEDs (both old and active)
+const posts = ref([])
 
-const infiniteIdWanteds = ref(1)
-watch(wanteds, () => infiniteIdWanteds.value++)
+const offersLoading = ref(true)
+const wantedsLoading = ref(true)
 
-const oldWantedCount = computed(() => {
-  return messages.filter(
-    (message) => message.type === 'Wanted' && message.hasoutcome
-  ).length
-})
+if (myid) {
+  offersLoading.value = true
+  wantedsLoading.value = true
 
-const oldWantedCountStr = computed(() => {
-  return pluralize('old WANTED', oldWantedCount.value, true)
-})
+  posts.value = await messageStore.fetchByUser(myid, false, true)
 
-const activeWantedCount = computed(() => {
-  return messages.filter(
-    (message) => message.type === 'Wanted' && !message.hasoutcome
-  ).length
-})
+  offersLoading.value = false
+  wantedsLoading.value = false
 
-const wantedsToShow = ref(1)
+  // No need to wait for searches - often below the fold.
+  searchStore.fetch(myid)
+}
 
-const wantedsShown = computed(() => {
-  return wanteds.value.slice(0, wantedsToShow.value)
+const shownOffersCount = ref(1)
+const offers = computed(() => {
+  return posts.value.filter((message) => message.type === 'Offer')
 })
 
-async function loadMoreWanteds($state) {
-  wantedsToShow.value++
+async function loadMoreOffers(infiniteLoaderInstance) {
+  shownOffersCount.value++
 
-  if (wantedsToShow.value > wanteds.value.length) {
-    wantedsToShow.value = wanteds.value.length
-    busyWanteds.value = false
-    $state.complete()
+  if (shownOffersCount.value > offers.value.length) {
+    shownOffersCount.value = offers.value.length
+
+    infiniteLoaderInstance.complete()
   } else {
-    await messageStore.fetch(wanteds.value[wantedsToShow.value - 1].id)
-    $state.loaded()
+    // only show the loading indicator when loading the active offers
+    if (!offers.value[shownOffersCount.value - 1].hasoutcome)
+      offersLoading.value = true
```

Comment:
> Do you mean that if I have loaded the active posts and then scrolled down faster than the network, then I wouldn't see any loading indicator?

I guess so, yes. But my idea was to solve another issue: when the button to show the old posts is not clicked (so that the old posts should not be visible) and you scroll down through active posts, when you reach the end, it's gonna show the loading indicator (loading the next posts, old ones), but then the loading spinner will disappear, but it won't be "substituted" by any posts (because what has been loaded is an old post and they're not visible).